apps/chronicles/hooks/useProgressUpload.ts

```diff
@@ -0,0 +1,100 @@
+import { ProgressStatus, ProgressTask, ProgressTaskErrorDetails } from "@equinor/mad-components";
+import { useState } from "react";
+import { Alert } from "react-native";
+
+type UploadScenario = "success" | "fail";
+
+const simulateTask = (duration: number, shouldFail = false) => {
+    return new Promise((resolve, reject) => {
+        setTimeout(() => {
+            if (shouldFail) {
+                reject(new Error("Upload failed"));
+            } else {
+                resolve("Upload successful");
+            }
+        }, duration);
+    });
+};
+
+export const useProgressUpload = () => {
+    const [tasks, setTasks] = useState<ProgressTask[]>([
+        { title: "CatWithHat1.jpg", status: "notStarted" },
+        { title: "CatWithHat2.jpg", status: "notStarted" },
+        { title: "CatWithHat3.jpg", status: "notStarted" },
+        { title: "CatWithHat4.jpg", status: "notStarted" },
+        { title: "CatWithHat5.jpg", status: "notStarted" },
+        { title: "CatWithHat6.jpg", status: "notStarted" },
+    ]);
+
+    const [isSimulating, setIsSimulating] = useState(false);
+
+    const updateTaskStatus = (
+        taskIndex: number,
+        status: ProgressStatus,
+        errorDetails?: ProgressTaskErrorDetails,
+        newTitle?: string,
+    ) => {
+        setTasks(tasks =>
+            tasks.map((task, index) =>
```

Comment:
I am getting some ESLint errors here regarding access on a `any` type. Should probably be fixed

Reply:
Nevermind. Reset and it seems to be gone now